### PR TITLE
Add sign in sign up

### PR DESCRIPTION
**Pull request by Malgorzataaa**
[BE] Add sign_in/sign_up (H)

### CODE REVIEW

File: .vs/ProjectSettings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": null
+}
```
Comment (h4b00) [inline .vs/ProjectSettings.json:3]: Add `.vs/*` to git ignore

File: .vs/VSWorkspaceState.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "\\config",
+    "\\db",
+    "\\spec",
+    "\\spec\\controllers\\api",
+    "\\spec\\controllers\\api\\v1",
+    "\\spec\\factories"
+  ],
+  "SelectedNode": "\\config\\environments",
+  "PreviewInSolutionExplorer": false
+}
```
Comment (h4b00) [inline .vs/VSWorkspaceState.json:16]: same as above

File: Gemfile [hunk 3]
```diff
@@ -27,6 +27,10 @@ gem 'rack-cors'
 
 gem 'active_model_serializers', '~> 0.10'
 
+gem 'devise'
+
+gem 'devise_token_auth'
```
Comment (h4b00) [inline Gemfile:32]: you need just one of them, and please add pessimistic version

File: app/controllers/api/v1/recipes_controller.rb [hunk 4]
```diff
@@ -1,7 +1,11 @@
 module Api
   module V1
     class RecipesController < ApplicationController
+
+      before_action :authenticate_user!
```
Comment (h4b00) [inline app/controllers/api/v1/recipes_controller.rb:5]: I am not sure if you want to block getting a list of recipes for anonymous users

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::API
+        include DeviseTokenAuth::Concerns::SetUserByToken
```
Comment (h4b00) [inline app/controllers/application_controller.rb:2]: indentation

File: db/migrate/20210819081137_devise_token_auth_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
+class DeviseTokenAuthCreateUsers < ActiveRecord::Migration[6.1]
+  
+  def change
+
```
Comment (h4b00) [inline db/migrate/20210819081137_devise_token_auth_create_users.rb:4]: unnecessary empty line

**Review by h4b00 — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 7]
```diff
@@ -27,6 +27,8 @@ gem 'rack-cors'
 
 gem 'active_model_serializers', '~> 0.10'
 
+gem 'devise_token_auth'
```
Comment (PatrykTyszka) [inline Gemfile:30]: version?

File: db/migrate/20210819081137_devise_token_auth_create_users.rb (hunk 6 above)
Comment (PatrykTyszka) [reply db/migrate/20210819081137_devise_token_auth_create_users.rb:3]: +1

File: db/migrate/20210819081137_devise_token_auth_create_users.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      t.boolean  :allow_password_change, :default => false
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Confirmable
+      t.string   :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (PatrykTyszka) [inline db/migrate/20210819081137_devise_token_auth_create_users.rb:24]: do you really need this?
look at the model
```
class User < ActiveRecord::Base
  extend Devise::Models
  # Include default devise modules. Others available are:
  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
  devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :validatable
  include DeviseTokenAuth::Concerns::User
end
```

File: db/schema.rb [hunk 9]
```diff
@@ -35,5 +35,30 @@
...
+    t.string "uid", default: "", null: false
+    t.string "encrypted_password", default: "", null: false
+    t.string "reset_password_token"
+    t.datetime "reset_password_sent_at"
+    t.boolean "allow_password_change", default: false
+    t.datetime "remember_created_at"
+    t.string "confirmation_token"
+    t.datetime "confirmed_at"
+    t.datetime "confirmation_sent_at"
+    t.string "unconfirmed_email"
```
Comment (PatrykTyszka) [inline db/schema.rb:49]: you should not have this here...

you need to rollback and run migrations once again if you want to change the schema.

OR you can clear it manually.

**Review by PatrykTyszka — CHANGES_REQUESTED** [review]

File: config/initializers/cors.rb [hunk 10]
```diff
@@ -7,7 +7,7 @@
 
 Rails.application.config.middleware.insert_before 0, Rack::Cors do
   allow do
-    origins ENV.fetch('API_ALLOWED_ORIGINS').to_s.split(',')
+    #origins ENV.fetch('API_ALLOWED_ORIGINS').to_s.split(',')
```
Comment (h4b00) [inline config/initializers/cors.rb:10]: why?

**Review by h4b00 — APPROVED** [review]
Just remove these `.vs` files

File: config/initializers/cors.rb (hunk 10 above)
Comment (Malgorzataaa) [reply config/initializers/cors.rb:10]: I shouldn't send you this file. Without '#' server doesn't work.

**Pull request closed without merging**